masschallenge/impact-api
# AC-6678: Allow a merge-to-deploy workflow to an AWS Stack e.g pre-staging

#### Changes introduced in [AC-6678](https://masschallenge.atlassian.net/browse/AC-6678)
- add logic that triggers an an ecs deploy when development is built

#### How to test
- rerunning the last [impact-api build](https://travis-ci.org/masschallenge/impact-api/builds/536757005?utm_source=github_status&utm_medium=notification) for this branch a deploy will be triggered at the end
- you can "see" the deployment in action on [ECS](https://console.aws.amazon.com/ecs/home?region=us-east-1#/clusters/pre-staging/services/testing-impact-service/tasks). you will notice that the re are now 2 tasks running showing that a deployment is in progress. The way deployments work, is a new task is created (to make a total of 2) and once that is healthy the old one is destroyed leaving the new task with the new code.
- also on the [events](https://console.aws.amazon.com/ecs/home?region=us-east-1#/clusters/pre-staging/services/testing-impact-service/events) section on ECS, you will notice a sequence of events that show a deployment took place. i.e.

```
service testing-impact-service has reached a steady state.
service testing-impact-service has stopped 1 running tasks: task a5d5b70b-f767-42e9-a444-97cdc367ad99.
service testing-impact-service has begun draining connections on 1 tasks.
service testing-impact-service deregistered 1 targets in target-group impact-pre-staging-target-group
service testing-impact-service registered 1 targets in target-group impact-pre-staging-target-group
service testing-impact-service has started 1 tasks: task 822f382f-f5b7-4606-8fc6-8ff4fb3c6eb3.
service testing-impact-service has reached a steady state.
```

ps, the logs are read from bottom to top
